src/colmap/scene/reconstruction.h

```diff
@@ -266,28 +266,46 @@ size_t Reconstruction::NumRegImages() const { return reg_image_ids_.size(); }
 size_t Reconstruction::NumPoints3D() const { return points3D_.size(); }
 
 const struct Camera& Reconstruction::Camera(const camera_t camera_id) const {
-  return cameras_.at(camera_id);
+  try {
+    return cameras_.at(camera_id);
+  } catch (const std::out_of_range& e) {
+    throw std::invalid_argument(
```

Comment:
The .at method throws an out of range exception. If we want full backwards compatibility on all call sites that may catch, we should rethrow the same.

Reply:
We don't catch it anywhere in the COLMAP codebase. But I agree that we should remain backward compatible with user code - I've switched to out-of-range, PTAL.